text/0328-polyglot-assert.md

```diff
@@ -0,0 +1,335 @@
+template.
+
+```ts
+inspect.assertMatchTemplateBeta1({
+  Resources: {
+    Type: 'Foo::Bar',
+    Properties: {
+      Baz: 'Qux',
+    },
+  },
+});
+```
+
+#### Counting Resources
+
+This module allows asserting the number of resources of a specific type found
+in a template.
+
+```ts
+inspect.assertResourceCountBeta1('Foo::Bar', 2);
+```
+
+#### Resource Matching
+
+Beyond resource counting, the module also allows asserting that a resource with
+specific properties are present.
+
+The following code asserts that the `Properties` section of a resource of type
+`Foo::Bar` contains the specified properties -
+
+```ts
+inspect.assertResourceBeta1('Foo::Bar', {
+  Foo: 'Bar',
+  Baz: 5,
+  Qux: [ 'Waldo', 'Fred' ],
+});
+```
+
+The same method allows asserting the complete definition of the 'Resource'
+which can be used to verify things other sections like `DependsOn`, `Metadata`,
+`DeletionProperty`, etc.
+
+```ts
+inspect.assertResourceBeta1('Foo::Bar', {
+  Properties: { Foo: 'Bar' },
+  DependsOn: [ 'Waldo', 'Fred' ],
+}, {
+  part: ResourcePartBeta1.COMPLETE_BETA1,
+});
+```
+
+## FAQ
+
+### What are we launching today?
+
+Today, we are launching a new jsii module called `@aws-cdk/assertv2`, that allows
+CDK users to run assertions on the synthesized CloudFormation template.
+
+This module is available to users in all CDK supported languages.
+
+### Why should I use this feature?
+
+You should use this feature if you would like to ensure that your CDK constructs
+and CDK applications render expected CloudFormation resources and properties.
+
+### How is this different from the existing `assert` module?
+
+The two modules provide the same set of features. However, the existing module
+was only available to javascript users.
+The new version launched today is available to users of all CDK supported
+languages available today and any that we will support in the future.
+
+### How do I get started?
+
+This module is available in the languages as per the table below -
+
+| Language              | Pkg Manager | Package                                     |
+| --------------------- | ------------|-------------------------------------------- |
+| typescript/javascript | npm         | `@aws-cdk/assertv2`                         |
+| Java                  | Maven       | `software.amazon.awscdk.assertv2`           |
+| Python                | PyPI        | `aws-cdk.assertv2`                          |
+| .NET                  | NuGet       | `Amazon.CDK.AssertV2`                       |
+| Go                    | -           | `github.com/aws/aws-cdk-go/awscdk/assertv2` |
+
+### Is this feature available in the AWS CDK v2?
+
+This module is available in both the AWS CDK v1 and v2, just like all of our other
+modules.
+
+In CDKv2, this module will be available as part of `aws-cdk-lib`.
+The import statement you will need to use is -
+
+```ts
+import { assertv2 as assert } from 'aws-cdk-lib';
```

Comment:
Why do we need to call this `assertv2` in `aws-cdk-lib`? There isn't an `assert` submodule there AFAIK. Let's clean that up in v2...

Reply:
Per Rico's suggestion, I'm going to call this `assertions` - https://github.com/aws/aws-cdk-rfcs/pull/330#discussion_r643943776. So that's what it will be in v2.

Reply:
Love it